File: be/mytown/Models/DbModels/PostReport.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace mytown.Models.DbModels
{
    public class PostReport
    {
        public int Id { get; set; }
        public int PostID { get; set; }
        public int SenderID { get; set; }
        public int ReportedUserID { get; set; }
        public DateTime Date { get; set; }
        public string Text { get; set; }
        public bool Read { get; set; }

        [ForeignKey("PostID")]
        public virtual Post Post { get; set; }
        [ForeignKey("SenderID")]
        public User Sender { get; set; }
        [ForeignKey("ReportedUserID")]
        public User ReportedUser { get; set; }
    }
}

File: fe/mobile_app/lib/models/DbModels/PostReport.dart
class PostReport
{
  int _postID;
  int _senderID;
  int _reportedUserID;
  DateTime _date;
  String _text;

  PostReport(this._postID, this._senderID, this._reportedUserID, this._date, this._text);

  Map<String, dynamic> toMap()
  {
    var data = Map<String, dynamic>();

    data["postID"] = this._postID;
    data["senderID"] = this._senderID;
    data["reportedUserID"] = this._reportedUserID;
    data["date"] = this._date.toIso8601String();
    data["text"] = this._text;

    return data;
  }
}